be/src/storage/lake/compaction_scheduler.cpp

```diff
@@ -223,21 +231,72 @@ void CompactionScheduler::thread_task(int id) {
     }
 }
 
+static bool compaction_should_cancel(CompactionTaskContext* context) {
+    if (context->callback->has_error() || context->callback->timeout_exceeded()) {
+        return true;
+    }
+
+#ifndef BE_TEST
+    int64_t check_interval_seconds = 60LL * config::lake_compaction_check_valid_interval_minutes;
+    if (!context->is_checker || check_interval_seconds <= 0) {
+        return false;
+    }
+
+    int64_t now = time(nullptr);
+    if (now > context->last_check_time && (now - context->last_check_time) >= check_interval_seconds) {
+        // ask FE whether this compaction transaction is still valid
+        TNetworkAddress master_addr = get_master_address();
+        if (master_addr.hostname.size() > 0 && master_addr.port > 0) {
+            TReportCloudCompactionRequest request;
+            request.__set_txn_id(context->txn_id);
+            TReportCloudCompactionResponse result;
+            auto status = ThriftRpcHelper::rpc<FrontendServiceClient>(
+                    master_addr.hostname, master_addr.port,
+                    [&request, &result](FrontendServiceConnection& client) {
+                        client->reportCloudCompaction(result, request);
+                    },
+                    3000 /* timeout 3 seconds */);
+            if (status.ok()) {
+                if (!result.valid) {
+                    // notify all tablets in this compaction request
+                    context->callback->update_status(Status::Aborted("compaction validation failed"));
+                    LOG(WARNING) << "validate compaction transaction " << context->txn_id << " for tablet "
+                                 << context->tablet_id << ", abort invalid compaction";
+                    return true; // should cancel compaction
+                } else {
+                    // everything is fine
+                }
+            } else {
+                LOG(WARNING) << "fail to validate compaction transaction " << context->txn_id << " for tablet "
```

Comment:
may consider LOG_N in case of FE service unavailable or BE network issue, causes massive log print in be logfile.

Reply:
if network error happens, it wil wait for next interval

Reply:
it is fine to wait, but how about all the tablets in compaction queue prints the logs at the same time.